addons/promote/ftests/src/main/java/org/commonjava/indy/promote/ftest/GroupPromoteThenHostedRepoTimeoutTest.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) 2011 Red Hat, Inc. (jdcasey@commonjava.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.promote.ftest;
+
+import org.commonjava.indy.model.core.ArtifactStore;
+import org.commonjava.indy.model.core.Group;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.commonjava.indy.promote.client.IndyPromoteClientModule;
+import org.commonjava.indy.promote.model.GroupPromoteRequest;
+import org.commonjava.indy.promote.model.GroupPromoteResult;
+import org.junit.Test;
+
+import static org.commonjava.indy.model.core.StoreType.hosted;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+public class GroupPromoteThenHostedRepoTimeoutTest
+        extends AbstractPromotionManagerTest
+{
+
+    @Test
+    public void run()
+            throws Exception
+    {
+        final int REPO_TIMEOUT_SECONDS = 6;
+        final int TIMEOUT_WAITING_MILLISECONDS = 8000;
+        final String hostedRepo = "hostedRepo";
+
+        final HostedRepository repo = new HostedRepository( hostedRepo );
+        repo.setRepoTimeoutSeconds( REPO_TIMEOUT_SECONDS );
+
+        client.stores().create( repo, "adding hosted", HostedRepository.class );
+        HostedRepository loaded = client.stores().load( hosted, hostedRepo, HostedRepository.class );
```

Comment:
FYI, return type of the call above this one returns the version of the HostedRepository as it was stored in Indy...so this load statement is a bit redundant.